baseframe/forms/validators.py

```diff
@@ -161,6 +162,51 @@ def compare(self, value, other):
         return value != other
 
 
+class IsPublicEmailDomain(object):
+    def __init__(self, message=None):
+        self.message = message or _(u'This domain is not a public email domain.')
+
+    def make_cache_key(self, email_or_domain):
+        return 'mxrecord/' + email_or_domain
```

Comment:
This function can be moved inline into `get_mx` as it's only called once.

Reply:
You may need the `six.PY2` check as used in `ValidUrl`. Add an Emoji domain to your tests to confirm the cache key works. Pick something from https://i❤.ws/ (https://xn--i-7iq.ws/)